### PR TITLE
Added Arecibo Observatory Site Data

### PR DESCRIPTION
**Pull request by Hazboun6**
I added in the Arecibo Observatory site information. I also cleaned up some of the later json entries to make them more uniform, i.e. to match the form of the other entries. 

I ran the `check_builtin_matches_remote` method, got a few errors and fixed them as can be seen in the commit notes. 

#pyastro18

```
"ao": {
        "aliases": [
            "AO",
            "Arecibo Observatory",
            "Arecibo",
            "arecibo"
            ],
        "elevation": 497,
        "elevation_unit": "meter",
        "latitude": 18.34416667,
        "latitude_unit": "degree",
        "longitude": 293.24722222,
        "longitude_unit": "degree",
        "name": "Arecibo Observatory",
        "source": "https://en.wikipedia.org/wiki/Arecibo_Observatory"
    }
```

### CODE REVIEW

**Comment by eteq** [comment]
Thanks @Hazboun6 - I'm guessing so based on the last commit... but just to be sure, can you confirm this validated correctly following the instructions here: https://github.com/astropy/astropy-data/blob/gh-pages/coordinates/README.md ?

**Comment by Hazboun6** [comment]
Hi @eteq, Yes I ran through the checks in the README. And they were useful.

**Comment by aarchiba** [comment]
Is there any reason not to merge this?

**Comment by hamogu** [comment]
It is very sad that your PR has outlived the telescope. And three years ago, you asked "is there any reason not to merge this". No , there is not - we were just missing a maintainer that takes time to look. The Astropy Community has clearly not done a good job with that, and I'm sorry. I'm rebasing this to resolve merge conflict and will merge it when that works.

File: coordinates/sites.json
Comment (hamogu) [inline coordinates/sites.json]:
```suggestion
    },
```

**Review by hamogu — APPROVED** [review]

**Pull request merged**
